### PR TITLE
10 detail district screen

### LINKED ISSUE #10: Detail District Screen

### PR DESCRIPTION
**Pull request by darot-chen**
Screen Shot 2021-07-14 at 12 13 03 AM
Screen Shot 2021-07-14 at 12 12 56 AM

### CODE REVIEW

**Comment by theachoem** [comment]
@darot-chen 

- please use a bit more padding on top and bottom on each tile. Material only gives us guidance and styling that work best for most English fonts. This tile looks very small with Khmer font. 
- use the lighter background for ភូមិ
- no border when tile is opened
- use default centerTitle (remove it from appBar or keep it null) for page transition.

**Comment by darot-chen** [comment]
> @darot-chen
> 
>     * please use a bit more padding on top and bottom on each tile. Material only gives us guidance and styling that work best for most English fonts. This tile looks very small with Khmer font.
> 
>     * use the lighter background for ភូមិ
> 
>     * no border when tile is opened
> 
>     * use default centerTitle (remove it from appBar or keep it null) for page transition.

Screen Shot 2021-07-14 at 9 00 04 AM

File: lib/screens/home/local_widget/province_card.dart [hunk 1]
```diff
@@ -72,23 +72,22 @@ class ProvinceCard extends StatelessWidget {
...
               (isKrong || isKhan)
                   ? ListTile(
                       title: Text((isKrong ? 'ក្រុង' : 'ខណ្ឌ') + (district[index].khmer ?? '')),
                       subtitle: Text('$sangkat និង $village'),
-                      onTap: () {
-                        Navigator.pushNamed(context, RouteConfig.DISTRICT);
-                      },
+                      onTap: onTap,
                     )
                   : ListTile(
```
Comment (theachoem) [inline lib/screens/home/local_widget/province_card.dart]: we should just update the title & subtitle here, so we don't have to write 2 widgets

eg. 
```dart
String title, subtitle;
if(isKrong || isKhan){title = "sth", subtitle = "sth"}
else {title = "1", subtitle = "2"}
```

File: lib/screens/district/district_screen.dart [hunk 2]
```diff
@@ -13,16 +17,114 @@ class DistrictScreen extends StatelessWidget {
...
+    List<TbCommuneModel> communes = geo.communesSearch(districtCode: district.code.toString());
+    String appBarTitle = (district.type == "SROK" ? 'ស្រុក' : 'ខណ្ឌ') + district.khmer.toString();
     return Scaffold(
       backgroundColor: Theme.of(context).colorScheme.background,
       body: CustomScrollView(
         slivers: [
           MorphingSliverAppBar(
             forceElevated: true,
-            title: CgAppBarTitle(title: district.khmer.toString()),
+            centerTitle: true,
```
Comment (theachoem) [inline lib/screens/district/district_screen.dart]: let us use default `centerTitle`, since it help improve animation when page transition.

**Review by theachoem — CHANGES_REQUESTED** [review]

**Review by theachoem — APPROVED** [review]

**Comment by theachoem** [comment]
@darot-chen  Cool

**Pull request merged**